src/Control/Monad/Aff/Internal.js

```diff
@@ -25,6 +25,22 @@ exports._takeVar = function (nonCanceler, avar) {
   };
 };
 
+exports._tryTakeVar = function (nothing, just, nonCanceler, avar) {
+  return function (success, error) {
+    if (avar.error !== undefined) {
+      error(avar.error);
+    } else if (avar.producers.length > 0) {
+      avar.producers.shift()(function (x) {
+        success(just(x));
+        return nonCanceler;
```

Comment:
I don't think this is necessary. Judging by the other function (which just pass in `success` and `error` to the producer, this should just be equivalent to `success <<< Just`. Which means the JS should be:
```js
avar.producers.shift()(function (x) {
  return success(just(x));
}, error);
```

Same for the `tryPeekVar` case.